### PR TITLE
chore: Push after auto commit

### LINKED ISSUE #29810: Push the commit that was added as part of the auto commit
After the auto commit, we need to push the commit if there is no change in the upstream branch. If there's a change, then the push should fail without any impact on

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
This PR pushes the commit added by the auto commit. 

#### PR fixes following issue(s)
Fixes #29810

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Manual
- [x] JUnit

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Enhanced auto-commit functionality with added support for private and public keys.
	- Improved DSL migration process with a new repository push step.

- **Bug Fixes**
	- Adjusted auto-commit progression steps and completion percentages for better user feedback.

- **Tests**
	- Updated tests to cover new authentication features and auto-commit logic.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/AutoCommitEventHandlerCEImpl.java [hunk 1]
```diff
@@ -150,6 +150,22 @@ public Mono<Boolean> autoCommitDSLMigration(AutoCommitEvent autoCommitEvent) {
...
+                                    autoCommitEvent.getApplicationId(),
+                                    autoCommitEvent.getRepoName());
+
+                            return gitExecutor
+                                    .pushApplication(
+                                            baseRepoSuffix,
+                                            autoCommitEvent.getRepoUrl(),
+                                            autoCommitEvent.getPublicKey(),
+                                            autoCommitEvent.getPrivateKey(),
+                                            autoCommitEvent.getBranchName())
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/AutoCommitEventHandlerCEImpl.java:166]: What happens when the push fails? We should handle this scenario with proper state. This method returns true even when the push fails

**Review by AnaghHegde — CHANGES_REQUESTED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/AutoCommitEventHandlerCEImpl.java (hunk 1 above)
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/AutoCommitEventHandlerCEImpl.java:166]: When the push fails, we're in a state where local head is ahead of remote. I've checked that scenario, there's nothing we need to do. The next time, when user is adding a new commit, there'll be a pull and push.

**Review by AnaghHegde — APPROVED** [review]

**Pull request merged**